src/net.cpp

```diff
@@ -2300,11 +2303,16 @@ bool CConnman::InitBinds(const std::vector<CService>& binds, const std::vector<N
     }
     if (binds.empty() && whiteBinds.empty()) {
         struct in_addr inaddr_any;
-        inaddr_any.s_addr = INADDR_ANY;
+        inaddr_any.s_addr = htonl(INADDR_ANY);
         struct in6_addr inaddr6_any = IN6ADDR_ANY_INIT;
         fBound |= Bind(CService(inaddr6_any, GetListenPort()), BF_NONE, NetPermissionFlags::PF_NONE);
         fBound |= Bind(CService(inaddr_any, GetListenPort()), !fBound ? BF_REPORT_ERROR : BF_NONE, NetPermissionFlags::PF_NONE);
     }
+
+    if (m_listen_onion) {
+        fBound |= Bind(onion_bind, BF_EXPLICIT, NetPermissionFlags::PF_NONE);
+    }
+
```

Comment:
For "127.0.0.1" `IsRoutable()` returns `false`, therefore it won't be passed to AddLocal(). But, in general, I agree with you.

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/19991#issuecomment-699005931).

Reply:
Yes, this is only relevant if a routable address is supplied to `-bind=addr:port=onion`.

The added `NetPermissionFlags::PF_ONION` works, but is strange because that is not a "permission". It is just a flag telling `Bind()` "this is a covert bind address, don't advertise it". Isn't `enum BindFlags` more suitable for it? `BF_DONT_ADVERTISE` or `BF_COVERT`?

In any case - the `_ONION` in the name may be too narrow, if we want to use the same mechanism for something else in the future.

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/19991#issuecomment-700831638).